text-generation-inference/Dockerfile

```diff
@@ -70,7 +72,15 @@ COPY server/pyproject.toml server/pyproject.toml
 RUN cd server && \
     make gen-server && \
     pip install -r requirements.txt && \
-    pip install . --no-cache-dir
+    pip install . --no-cache-dir && \
+    pip install git+https://github.com/HabanaAI/DeepSpeed.git@1.13.0 && \
+    pip install git+https://github.com/huggingface/optimum-habana.git && \
+    pip install transformers==4.34.1
```

Comment:
I don't think we should install transformers here (it's 3rd time: in requirements.txt, in pyproject.toml and now).
I would suggest to take a look on pyproject.toml -- due to this file transformers package is replaced with the older version.

Reply:
Yes it shouldn't be done here.
Every dependencies (for DeepSpeed it's different since it's installed directly from a GH repo) should actually only be specified in pyproject.toml. Then, we can automatically generate requirements.txt with Poetry.

Reply:
when i tried to update in the pyproject.toml to 4.34.1 , it gives me conflicts between tgi 1.1.0 and transformer version. both uses different tokenizer versions.
any suggestions ?

Reply:
Ah yeah I see. Do we absolutely need TGI v1.1 at the moment? Or would v1.0 be okay and not impact this PR too much?